Backend/app/dashboard/Lib/src/Tools/Auth/Common.hs

```diff
@@ -44,22 +44,37 @@ verifyPerson ::
 verifyPerson token = do
   key <- authTokenCacheKey token
   authTokenCacheExpiry <- getSeconds <$> asks (.authTokenCacheExpiry)
+
+  -- First check if token exists in cache
   mbTuple <- getKeyRedis key
   (personId, merchantId, city) <- case mbTuple of
-    Just (personId, merchantId, city) -> return (personId, merchantId, city)
+    Just (personId, merchantId, city) -> do
+      -- For cached tokens, check activity in Redis
+      now <- getCurrentTime
+      checkTokenActivityInRedis token now
+      return (personId, merchantId, city)
     Nothing -> do
       mbTupleOld <- getKeyRedisOld key
       case mbTupleOld of
         Just (personId, merchantId) -> do
+          -- For old cache format, check activity in Redis
+          now <- getCurrentTime
+          checkTokenActivityInRedis token now
           city <- getCity merchantId
           return (personId, merchantId, city)
         Nothing -> do
+          -- For tokens not in cache, verify from DB
           sr <- verifyToken token
```

Comment:
what if this function will throw error? then old keys from `tokenActivityCacheKey` will not be removed